src/twisted/_version.py

```diff
@@ -7,5 +7,5 @@
 
 from incremental import Version
 
-__version__ = Version('Twisted', 17, 1, 0, dev=0)
+__version__ = Version('Twisted', 17, 1, 1, dev=0)
```

Comment:
Consider any of these version constraints:

  * `twisted==17.1.*`
  * `twisted=~=17.1`
  * `twisted>=17.1,==17.*`

One might expect any of these to be satisfied by any of the 17.1.* releases (17.1.0, 17.1.1, 17.1.2, etc).  One might expect it _not_ to include any changes added to trunk@HEAD after the 17.1.x release branch is created.

However, if trunk@HEAD declares itself as `17.1.1.dev0` then it falls within the declared range.  It would be surprising and unfortunate to end up with the wrong version in this case.  And since it is _so_ easily avoided I don't see why we would choose to do it this way.

The 17.1 release branch is the branch that is going to continue on to 17.1.1.  Marking _that_ branch as 17.1.1.dev0 seems unambiguously correct.

The trunk branch is the branch that is going to continue in to `17.(Y>1)` (or possibly `(X>17).Y` if there's some catastrophe).  The problem that it is difficult to predict what X and Y are going to be (aside from knowing the minimum possible values) doesn't seem like a justification to screw up things for the 17.1.x line.

So, please don't put this version into trunk.  Put it into the release branch.  Put 17.3.0.dev0 into trunk.  Or 17.4.0.dev0 if that seems more realistic.  And if it turns out to have the wrong Y component - oh well, that's kind of unavoidable with an unpredictable versioning scheme like the one used by Twisted.  Practice harder to predict the next release more accurately, I guess.  Or let go of the idea that `postN` is only for metadata changes.  But, again, none of that is reason to screw up the 17.1.x line.